relay-common/src/time.rs

```diff
@@ -66,19 +62,11 @@ impl fmt::Debug for UnixTimestamp {
     }
 }
 
-impl std::ops::Add<Duration> for UnixTimestamp {
-    type Output = Self;
-
-    fn add(self, rhs: Duration) -> Self::Output {
-        Self(self.0 + rhs)
-    }
-}
-
 impl std::ops::Sub for UnixTimestamp {
     type Output = Duration;
 
     fn sub(self, rhs: Self) -> Self::Output {
-        self.0 - rhs.0
+        Duration::from_secs(self.0 - rhs.0)
```

Comment:
Wouldn't this overflow if `rhs.0` > `self.0`? Might be an idea to represent the timestamp as `i64` instead.

Reply:
The standard library `Duration` cannot be negative, and we also do not aim to support negative unix timestamps.
It's worth noting that the `Sub` has undefined behavior for all unsigned numeric types.

I'm much less happy with the `UnixTimestamp` type now that it lost subsecond precision. However, the general expectation of a unix timestamp is to be in whole seconds, hence the change. This is also what caused the bug in the first place.